src/spu/src/services/public/stream_fetch.rs

```diff
@@ -312,103 +302,100 @@ impl StreamFetchHandler {
-                            &self.header,
-                            filter_response,
-                        );
+            let consumer_wait = if !filter_batch.records().is_empty() {
+                trace!("filter batch: {:#?}", filter_batch);
 
-                    trace!("sending back filter respone: {:#?}", filter_response_msg);
+                next_offset = filter_batch.get_last_offset() + 1;
 
-                    let mut inner_sink = self.sink.lock().await;
-                    inner_sink
-                        .send_response(&filter_response_msg, self.header.api_version())
-                        .await?;
-                    true
-                } else {
-                    debug!(next_offset, "filter, no records send back, skipping");
-                    false
+                debug!(
+                    next_offset,
+                    records = filter_batch.records().len(),
+                    "sending back to consumer"
+                );
+                let records = RecordSet::default().add(filter_batch);
+                let filter_partition_response = DefaultPartitionResponse {
+                    partition_index: self.replica.partition,
+                    error_code: file_partition_response.error_code,
+                    high_watermark: file_partition_response.high_watermark,
+                    log_start_offset: file_partition_response.log_start_offset,
+                    records,
+                    next_filter_offset: next_offset,
+                    // we mark last offset in the response that we should sync up
+                    ..Default::default()
                 };
 
-                Ok((next_offset, consumer_wait))
-            } else {
-                debug!("no filter, sending back entire");
-
-                let response = StreamFetchResponse {
+                let filter_response = StreamFetchResponse {
                     topic: self.replica.topic.clone(),
                     stream_id: self.stream_id,
-                    partition: file_partition_response,
+                    partition: filter_partition_response,
                 };
 
-                let response_msg = RequestMessage::<FileStreamFetchRequest>::response_with_header(
-                    &self.header,
-                    response,
-                );
+                let filter_response_msg =
+                    RequestMessage::<DefaultStreamFetchRequest>::response_with_header(
+                        &self.header,
+                        filter_response,
+                    );
 
-                trace!("sending back file fetch response msg: {:#?}", response_msg);
+                trace!("sending back filter respone: {:#?}", filter_response_msg);
 
                 let mut inner_sink = self.sink.lock().await;
                 inner_sink
-                    .encode_file_slices(&response_msg, self.header.api_version())
+                    .send_response(&filter_response_msg, self.header.api_version())
```

Comment:
This diff is also misleading. This call to `inner_sink.send_response` corresponds to the following call in the original:

https://github.com/infinyon/fluvio/pull/1188/files#diff-5bcc5ec4673d6a5c8633195a7360c3402f873769c1faa90cf627b242179d2241L371-L373